src/beamlime/applications/_nexus_helpers.py

```diff
@@ -0,0 +1,303 @@
+            self.config_container['values'] = values
+        elif isinstance(self.config_container['values'], list):
+            self.config_container['values'].extend(values)
+
+
+class ModularDataGroupContainer:
+    """Module container protocol."""
+
+    def __init__(self, group_dict: dict, module_dict: dict, **initial_values) -> None:
+        self.sub_datasets: dict[str, ModularDatasetContainer]
+        self.module_dict = module_dict
+        self.group_dict = group_dict
+        self._populate_sub_dataset()
+        self.update(initial_values)
+
+    def _populate_sub_dataset(self) -> None:
+        """Populate the sub datasets from the module dictionary."""
+        module_id = self.module_dict['module']
+        dataset_recipes = MODULE_RECIPES[module_id]
+        self.group_dict['children'].remove(self.module_dict)
+        self.sub_datasets = {
+            name: ModularDatasetContainer(name, tp)
+            for name, tp in dataset_recipes.items()
+        }
+        self.group_dict['children'].extend(self.sub_datasets.values())
+
+    def update(self, other: dict) -> None:
+        """Update the sub datasets from the other. Ignore if the name is not present.
+
+        See :func:`~ModularDatasetContainer.update` for more details.
+        """
+        for name, sub_dataset in self.sub_datasets.items():
+            if name in other:
+                sub_dataset.update(other[name])
+
+    def clear(self) -> None:
+        """Clear the sub datasets.
+
+        See :func:`~ModularDatasetContainer.clear` for more details.
+        """
+        for sub_dataset in self.sub_datasets.values():
+            sub_dataset.clear()
+
+
+class NXDetectorContainer:
+    """NXDetector container.
+
+    This container is only for fake event data feeding.
+    """
+
+    def __init__(self, detector_dict: dict) -> None:
+        self.pixel_ids: list[int]
+        self.pixel_offsets: dict[str, list[float]]
+
+        self.detector_name: str = detector_dict['name']
+        self.detector_dict = detector_dict
+
+        self.init_pixel_ids()
+        self.init_pixel_offsets()
+
+    def init_pixel_ids(self) -> None:
+        """Initialize the pixel ids of the detector."""
+        pixel_id_filter = partial(_match_dataset_name, name='detector_number')
+        self._pixel_ids_idx = find_index(
+            self.detector_dict, 'children', pixel_id_filter, 'config', 'values'
+        )
+        self.pixel_ids = nested_dict_getitem(self.detector_dict, *self._pixel_ids_idx)
+
+    def init_pixel_offsets(self) -> None:
+        """Initialize the pixel offsets of the detector."""
+        pixel_offsets = [
+            child
+            for child in self.detector_dict['children']
+            if _match_dataset_name_suffix(child, 'pixel_offset')
+        ]
+        self.pixel_offsets = {
+            offset['config']['name']: offset['config']['values']
+            for offset in pixel_offsets
+        }
+        self._pixel_offsets_idx = {
```

Comment:
What is this private field used for?

Reply:
Just for pruning them later...?

Reply:
Later where?

Reply:
In the ``prune_pixel_info``, which is deleted so this part is also deleted.